plasma_framework/test/endToEndTests/FeeClaim.e2e.test.js

```diff
@@ -0,0 +1,263 @@
+            );
+
+            this.framework.addExitQueue(config.registerKeys.vaultId.eth, ETH);
+        });
+
+        describe('And then first fee collected when Alice transfer to Bob', () => {
+            describe('And then second fee collected when Alice transfer to Carol', () => {
+                describe('When operator spends the fee outputs to Payment transaction', () => {
+
+                });
+            });
+        });
+
+        describe('When Alice deposits ETH to the plasma', () => {
+            let alicePlasmaBalance;
+            let aliceDepositUtxoPos;
+
+            before(async () => {
+                const depositBlockNum = (await this.framework.nextDepositBlock()).toNumber();
+                aliceDepositUtxoPos = buildUtxoPos(depositBlockNum, 0, 0);
+
+                const depositTx = Testlang.deposit(PAYMENT_OUTPUT_TYPE, DEPOSIT_VALUE, alice);
+                alicePlasmaBalance = DEPOSIT_VALUE;
+
+                return this.ethVault.deposit(depositTx, { from: alice, value: DEPOSIT_VALUE });
+            });
+
+            describe('When Alice transfer to Bob, fee is implicitly collected', () => {
+                let aliceBalanceUtxoPosAfterTransferToBob;
+                let aliceTransferToBobTxBytes;
+
+                before(async () => {
+                    const transferAmount = 1000;
+                    alicePlasmaBalance -= (transferAmount + FEE_AMOUNT);
+                    const outputBob = new PaymentTransactionOutput(PAYMENT_OUTPUT_TYPE, transferAmount, bob, ETH);
+                    const outputAlice = new PaymentTransactionOutput(
+                        PAYMENT_OUTPUT_TYPE, alicePlasmaBalance, alice, ETH,
+                    );
+
+                    const aliceBalanceOutputIndex = 1;
+                    const blockNum = (await this.framework.nextChildBlock()).toNumber();
+                    aliceBalanceUtxoPosAfterTransferToBob = buildUtxoPos(blockNum, 0, aliceBalanceOutputIndex);
+
+                    const txObj = new PaymentTransaction(
+                        PAYMENT_TX_TYPE,
+                        [aliceDepositUtxoPos],
+                        [outputBob, outputAlice],
+                    );
+                    aliceTransferToBobTxBytes = web3.utils.bytesToHex(txObj.rlpEncoded());
+                });
+
+                describe('And then operator mined the block with the first fee tx claiming the fee', () => {
+                    let firstFeeTxBytes;
+                    let firstFeeClaimUtxoPos;
+                    let firstFeeTxMerkleProof;
+
+                    before(async () => {
+                        const nextBlockNum = (await this.framework.nextChildBlock()).toNumber();
+                        const feeOutputs = [
+                            new FeeClaimOutput(FEE_OUTPUT_TYPE, FEE_AMOUNT, operatorFeeAddress, ETH),
+                            new FeeBlockNumOutput(FEE_NONCE_OUTPUT_TYPE, nextBlockNum),
```

Comment:
Wasn't it agreed that block number holding output should be first?
https://github.com/omisego/feat-fee/blob/master/docs/fee_exit_design.md#transaction-design

Reply:
oh...awwwww. anyway, probably I would wait a bit since we might remove the second outputs as proposed in: https://github.com/omisego/feat-fee/pull/47